### ISSUE TITLE
How to remove Weight_Normalization?

### ISSUE THREAD

**Issue by MachineJeff** [tensorflow/addons#763]
## Question

Hi !

I have successfully applied the **weight normalization wrapper** in my model, and the model can run without any bug.

However, I wanna know how to remove the weight normalization at **inference** time?

Just like the **layer normalization**，as we all know,  **layer normalization** API has a param called **is_training**, so that **layer normalization** can be removed automatically.

In a project written with pytorch called [melgan ](https://github.com/seungwonpark/melgan) 

I read their pytorch code and I found that they remove weight norm at inference time. What's more, I have pulled an [issue](https://github.com/seungwonpark/melgan/issues/37) in that project.
![iuuse](https://user-images.githubusercontent.com/30699922/70790005-14a4c580-1dcf-11ea-9e2d-a0b2225ba345.PNG)

## Key code in my model

```python
def wn_conv1d(x, kernel_size, channels, scope, stride=1, pad='same', dilation=1, groups=1):
  with tf.variable_scope(scope):
    output = WeightNorm(group_Conv1D(
                          filters = channels, 
                          kernel_size = kernel_size, 
                          strides = stride, 
                          padding = pad, 
                          dilation_rate = dilation,
                          groups = groups))(x)
    return output
```

## More

The **WeightNorm** is from this [repository](https://github.com/seanpmorgan/tf-weightnorm)

Haha, I found the contributor is @seanpmorgan , you sir are a hero!

In my opinion, addons should be inserted into tensorflow rather than left as a lone repository...

**Comment by seanpmorgan** [comment]
Hi @MachineJeff thanks for the compliment, but in reality the  WeightNorm in Addons has undergone many changes (from ~4 contributors) and is a better implementation. 

This request is valid though, and we should be able to support it.

**Comment by MachineJeff** [comment]
Haha, you all are heroes!

Actually, weight norm is removed in the project  [melgan ](https://github.com/seungwonpark/melgan) I mentioned above at inference time, so they get a good result. But I can not remove it so I get a not very good result.

Looking forward your support!.

**Comment by MachineJeff** [comment]
@seanpmorgan 
Hi, I have more discoveries.

I apply WeightNorm in **Group_Convolution**.

But **Group_Convolution** have not only one Kernel Weight

Maybe WeightNorm  has some bug dealing with **Group_Convolution**

**Group_Convolution** in [this PR](https://github.com/tensorflow/tensorflow/pull/10482/files)

**Comment by Squadrick** [comment]
@MachineJeff The PR you mentioned was never merged. See https://github.com/tensorflow/tensorflow/pull/25818 for group conv

**Comment by nukes** [comment]
again, how to remove weight norm ?

**Comment by bhack** [comment]
@nukes Can you see the test at https://github.com/tensorflow/addons/pull/819/files
